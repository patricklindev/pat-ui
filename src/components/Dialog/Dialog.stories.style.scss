.dialog-content input {
    font: inherit;
    padding: 4px 0px 5px;
    border: 0px;
    box-sizing: content-box;
    height: 1.4375em;
    margin-top: 10px;
    display: block;
    width: 100%;
    border-bottom: 1px solid black;
  }
  .dialog-btn {
    width: 85px;
    height: 24px;
  
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 24px;
  
    letter-spacing: 0.4px;
    text-transform: uppercase;
    color: #1976d2;
    border: white;
    background-color: transparent;
  
    &.hover {
      background-color: #1976d215;
    }
  }
  .email-row {
    margin-bottom: 20px;
    width: 105%;
    cursor: pointer;
  
    &.hover {
      background-color: #ddd;
    }
  }
  .dialog-actions-simple-row {
    cursor: pointer;
    padding: 20px 24px;
  
    &.hover {
      background-color: #ddd;
    }
  }
  
  span {
    display: inline-block;
    width: 60%;
    margin-left: 10px;
  }