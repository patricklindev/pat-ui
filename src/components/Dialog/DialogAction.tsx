import React, { Children, FC, ReactNode } from 'react';

export interface IDialogActionProps {
  className?: string;
  showCancelBtn?: boolean;
  showSubmitBtn?: boolean;
  cancelBtnTitle?: string;
  submitBtnTitle?: string;
  children?: ReactNode;
  onCancel?: (val?: any) => void;
  onSubmit?: (val?: any) => void;
}

const DialogAction: FC<IDialogActionProps> = (props) => {
  const {
    className,
    onCancel,
    onSubmit,
    showCancelBtn,
    showSubmitBtn,
    cancelBtnTitle,
    submitBtnTitle,
    children,
    ...rest
  } = props;

  let styleClasses = 'dialog-action';

  if (className) {
    styleClasses += ' ' + className;
  }

  // const handleCancel = (val?: any) => {
  //   if (!!onCancel) {
  //     onCancel(val);
  //     console.log('close triggered');
  //     console.log(onCancel);
  //   }
  // };

  // const handleSubmit = (val?: any) => {
  //   if (!!onSubmit) {
  //     onSubmit(val);
  //     console.log('submit triggered');
  //   }
  //   handleCancel();
  // };

  let cancelBtn;

  if (showCancelBtn) {
    cancelBtn = (
      <button className={styleClasses + ' btn'} onClick={onCancel}>
        {cancelBtnTitle}
      </button>
    );
  } else {
    cancelBtn = <></>;
  }

  let submitBtn;

  if (showSubmitBtn) {
    submitBtn = (
      <button className={styleClasses + ' btn'} onClick={onSubmit}>
        {submitBtnTitle}
      </button>
    );
  } else {
    submitBtn = <></>;
  }

  return (
    <div className={styleClasses}>
      {!!children ? (
        children
      ) : (
        <>
          {cancelBtn}
          {submitBtn}
        </>
      )}
    </div>
  );
};

DialogAction.defaultProps = {
  showCancelBtn: true,
  showSubmitBtn: true,
  cancelBtnTitle: 'cancel',
  submitBtnTitle: 'submit'
};

export default DialogAction;
