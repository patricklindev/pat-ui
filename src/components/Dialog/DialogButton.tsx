import React from 'react';

const Button = props => {
    return <button onClick={props.onClick}>
        {props.children}
    </button>;
};

export default Button;