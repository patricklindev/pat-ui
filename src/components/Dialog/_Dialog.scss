@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500&display=swap');
*{
    margin:0;
    padding:0;
    box-sizing: border-box;
}

.dialog{
    position: relative;
    width: 100vw;
    height: 100vh;
    font-family: "Roboto" 
}
.dialog>div{
  position: absolute;
  left:50%;
  top:50%;
  transform: translate(-50%, -50%);
  z-index: 2;
}

.dialog-btn{
    font-weight: 500;
    font-size: 0.875rem;
    line-height: 1.75;
    letter-spacing: 0.02857em;
    text-transform: uppercase;
    min-width: 64px;
    padding: 5px 15px;
    border-radius: 4px;
    border: 1px solid rgba(25, 118, 210, 0.5);
    color: #1976d2;
    background-color: white;
    cursor: pointer;
}
.dialog-btn:hover{
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
  background-color: transparent;
}

.overlay {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background: rgba(0, 0, 0, 0.5);
    transition: opacity 500ms;
    opacity: 1;
    z-index: 1;
  }
  .overlay-display{
    display: none;
  }
  // Several different types of dialogue
  .dialog-container-simple{
    width: 252px;
    height: 248px;
    border-radius: 4px;
    background-color: white;
  }

 .dialog-container-alert{
    width: 600px;
    height: 180px;
    border-radius: 4px;
    background-color: white;
 }
 .dialog-container-form{
    width: 600px;
    height: 245px;
    border-radius: 4px;
    background-color: white;
}

.dialog-header{
  width: 100%;
  height: 64px;
  position: relative;
}

.dialog-title{
      position: absolute;
      top:16px;
      left: 24px;
      font-style: normal;
      font-weight: 500;
      font-size: 20px;
      line-height: 160%;
      letter-spacing: 0.15px;
  }

 .dialog-content{
   width: 100%;
   height: 45%;
   padding: 0px 24px;
   font-size: 16px;
   font-weight: 400;
   line-height: 24px;
   letter-spacing: 0.15px;
   color: rgba(0, 0, 0, 0.6);
 }
 

 .dialog-content input{
  font: inherit;
  letter-spacing: inherit;
  color: currentcolor;
  padding: 4px 0px 5px;
  border: 0px;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0px;
  display: block;
  min-width: 0px;
  width: 100%;
  animation-name: mui-auto-fill-cancel;
  animation-duration: 10ms;
 }

 .dialog-actions{
  padding: 0 24px;
  font-size: 16px;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0.15px;
  color: rgba(0, 0, 0, 0.6);
 }
 

 