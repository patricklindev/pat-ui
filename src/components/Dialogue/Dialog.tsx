import React, { FC, MouseEvent } from 'react';
import { classNames } from '../../utils/classNames';

// export enum ButtonSize {
//   Large = 'lg',
//   Small = 'sm',
// }
export type ButtonSize = 'lg' | 'sm';
export type ButtonType =
  | 'primary'
  | 'secondary'
  | 'danger'
  | 'default'
  | 'link';
// export enum ButtonType {
//   Primary = 'primary',
//   Secondary = 'secondary',
//   Danger = 'danger',
//   Default = 'default',
//   Link = 'link',
// }
export interface IDialogProps {
  /** set customized style */
  className?: string;
}

type NativeButtonProps = IDialogProps;
type NativeAchorButtonProps = IDialogProps;
export type PatDialogProps = NativeButtonProps | NativeAchorButtonProps;

/**
 * A Button indicates a possible user action.
 *
 * ```js
 * import {Button} from 'pat-ui'
 * ```
 */
export const Dialog: FC<PatDialogProps> = (props) => {
  const { children, className, ...rest } = props;
  let styleClasses = 'dialog';
  if (className) styleClasses += ' ' + className;

  let dialog;
  dialog = <div className={styleClasses}>{props.children}</div>;
  return dialog;
};

Dialog.defaultProps = {};

export default Dialog;
