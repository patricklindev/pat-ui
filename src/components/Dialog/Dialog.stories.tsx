import React, { useState, useCallback } from 'react';
import { action } from '@storybook/addon-actions';
import Dialog from './index';
import Avatar from '@mui/material/Avatar';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import PersonIcon from '@mui/icons-material/Person';
import AddIcon from '@mui/icons-material/Add';
import { blue } from '@mui/material/colors';
import Input from '../Input';
import Button from '../Button';

export default {
  title: 'Dialog',
  component: Dialog,
};

const emails = ['username@gmail.com', 'user02@gmail.com'];

const buttonStyle: React.CSSProperties = {
  marginRight: '5px',
  marginTop: '5px',
};

export const SimpleDialogWithChildrenAndDraggable = () => {
  // The actions addon is used to display data received by event handler (callback) arguments in your stories.
  // lines: 25-32 provided by the user to control the dialog
  const [showModal, setModalState] = useState(false);
  const handleOpen = useCallback(() => {
    setModalState((prevState) => !prevState);
  }, []);
  const handleClose = (inputValue: string) => {
    action(inputValue);
    setModalState(false);
  };

  return (
    <>
      <Button style={buttonStyle} btnSize="sm" onClick={handleOpen}>
        click here for button
      </Button>

      <Dialog open={showModal} onClose={handleClose} draggable>
        <h5 className="title">Set backup account</h5>
        <List sx={{ pt: 0 }}>
          {emails.map((email) => (
            <ListItem button onClick={action(email)} key={email}>
              <ListItemAvatar>
                <Avatar sx={{ bgcolor: blue[100], color: blue[600] }}>
                  <PersonIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={email} />
            </ListItem>
          ))}
          <ListItem autoFocus button onClick={action('add account')}>
            <ListItemAvatar>
              <Avatar>
                <AddIcon />
              </Avatar>
            </ListItemAvatar>
            <ListItemText primary="Add account" />
          </ListItem>
        </List>
      </Dialog>
    </>
  );
};

export const AlertDialog = () => {
  const [showModal, setModalState] = useState(false);
  const handleOpen = useCallback(() => {
    setModalState((prevState) => !prevState);
  }, []);
  const handleClose = (inputValue?: string) => {
    setModalState(false);
  };

  return (
    <div>
      <button
        className={`${showModal ? 'blur' : ''}`}
        onClick={() => {
          //   action('button clicked');
          handleOpen();
        }}
      >
        Click here for Dialog
      </button>
      <Dialog open={showModal} onClose={handleClose}>
        <h5 className="title">Use Google's location service?</h5>
        <p className="modal__paragraph">
          Let Google help apps determine location. This means sending anonymous
          location data to Google, even when no apps are running.
        </p>
        <section className="modal-button__section">
          <button
            className="modal-button"
            onClick={() => {
              setModalState(false);
            }}
          >
            Cancel
          </button>
          <button className="modal-button" onClick={action('subscribed')}>
            Subscribe
          </button>
        </section>
      </Dialog>
    </div>
  );
};

export const SubscribeDialog = () => {
  const [showModal, setModalState] = useState(false);
  const handleOpen = useCallback(() => {
    setModalState((prevState) => !prevState);
  }, []);
  const handleClose = (inputValue?: string) => {
    setModalState(false);
  };
  const [email, setEmail] = useState<string>('');

  return (
    <div>
      <button
        className={`${showModal ? 'blur' : ''}`}
        onClick={() => {
          //   action('button clicked');
          handleOpen();
        }}
      >
        Click here for Dialog
      </button>
      <Dialog open={showModal} onClose={handleClose}>
        <h5 className="title">Subscribe</h5>
        <p className="modal__paragraph">
          To subscribe to this website, please enter your email address here. We
          will send updates occasionally
        </p>
        <Input
          onChange={(e) => {
            setEmail(e.target.value as string);
          }}
          placeholder="Email Here ..."
        />
        <section className="modal-button__section">
          <button
            className="modal-button"
            onClick={() => {
              handleClose();
            }}
          >
            Cancel
          </button>
          <button className="modal-button" onClick={action(email as string)}>
            Subscribe
          </button>
        </section>
      </Dialog>
    </div>
  );
};

export const DialogWithFullScreen = () => {
  const [showModal, setModalState] = useState(false);
  const handleOpen = useCallback(() => {
    setModalState((prevState) => !prevState);
  }, []);
  const handleClose = (inputValue: string) => {
    action(inputValue);
    setModalState(false);
  };

  return (
    <>
      <Button style={buttonStyle} btnSize="sm" onClick={handleOpen}>
        click here for button
      </Button>

      <Dialog open={showModal} onClose={handleClose} fullScreen>
        <h5 className="title">Use Google's location service?</h5>
        <p className="modal__paragraph">
          Let Google help apps determine location. This means sending anonymous
          location data to Google, even when no apps are running.
        </p>
        <section className="modal-button__section">
          <button
            className="modal-button"
            onClick={() => {
              setModalState(false);
            }}
          >
            Cancel
          </button>
          <button className="modal-button" onClick={action('subscribed')}>
            Subscribe
          </button>
        </section>
      </Dialog>
    </>
  );
};

export const DialogWithCustomStyling = () => {
  const [showModal, setModalState] = useState(false);
  const handleOpen = useCallback(() => {
    setModalState((prevState) => !prevState);
  }, []);
  const handleClose = (inputValue: string) => {
    action(inputValue);
    setModalState(false);
  };

  const customStyling: React.CSSProperties = {
    backgroundColor: 'yellow',
    display: 'flex',
    justifyContent: 'flex-end',
  };

  return (
    <>
      <Button style={buttonStyle} btnSize="sm" onClick={handleOpen}>
        click here for button
      </Button>

      <Dialog open={showModal} onClose={handleClose} style={customStyling}>
        <h5 className="title">Use Google's location service?</h5>
        <p className="modal__paragraph">
          Let Google help apps determine location. This means sending anonymous
          location data to Google, even when no apps are running.
        </p>
        <section className="modal-button__section">
          <button
            className="modal-button"
            onClick={() => {
              setModalState(false);
            }}
          >
            Cancel
          </button>
          <button className="modal-button" onClick={action('subscribed')}>
            Subscribe
          </button>
        </section>
      </Dialog>
    </>
  );
};
