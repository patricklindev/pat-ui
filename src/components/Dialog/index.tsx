import { FC } from 'react';
import Dialog, { IDialogProps } from './Dialog';
import DialogTitle, { IDialogTitleProps } from './DialogTitle';
import DialogContent, { IDialogContentProps } from './DialogContent';
import DialogAction, { IDialogActionProps } from './DialogAction';

export type PatDialogComponent = FC<IDialogProps> & {
  Title: FC<IDialogTitleProps>;
  Content: FC<IDialogContentProps>;
  Action: FC<IDialogActionProps>;
};

const PatDialog = Dialog as PatDialogComponent;
PatDialog.Title = DialogTitle;
PatDialog.Content = DialogContent;
PatDialog.Action = DialogAction;

export default PatDialog;