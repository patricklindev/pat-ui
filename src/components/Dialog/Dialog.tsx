import React, {
  ButtonHTMLAttributes,
  AnchorHTMLAttributes,
  FC,
  useState,
} from 'react';
import Button from './DialogButton';
import Modal from './DialogWindow';
import './_Dialog.scss';
import Icon from '../Icon/Icon'


export interface IButtonProps {
  /** set open/close style */
  open?: boolean;
  /** set window full size */
  fullScreen?: boolean;
  /** set scroll option  */
  scroll?: boolean;
}



type NativeButtonProps = IButtonProps & ButtonHTMLAttributes<HTMLButtonElement>;
type NativeAchorButtonProps = IButtonProps &
  AnchorHTMLAttributes<HTMLAnchorElement>;
export type PatButtonProps = NativeButtonProps | NativeAchorButtonProps;

/**
 * A Dialog is overlaid modal paper based components with a backdrop.
 *
 * ```js
 * import {Dialog} from 'pat-ui'
 * ```
 */
export const Dialog: FC<PatButtonProps> = (props) => {
const [showModal, setShowModal] = useState(false);

const ModalHeader = props => {
    return <div className="modal__header">
        {props.children}
    </div>
}

const ModalBody = props => {
    return <div className="modal__body">
        {props.children}
    </div>
}

    return (
        <div>
            <Button onClick={() => setShowModal(true)}>
                Open Dialog
            </Button>
            <Modal
                show={showModal}
                setShow={setShowModal}
            >
                <ModalHeader>
                    <div>Set backup account</div>
                </ModalHeader>
                <ModalBody>
                    <div className="item" onClick={() => setShowModal(false)}>
                        <Icon
      color="grey"
      disabled={false}
      loading={false}
      name="users"
      size="small"
    />&nbsp; &nbsp; user@gmail.com
                    </div>
                       <div className="item" onClick={() => setShowModal(false)}>
                        <Icon size="small"
      color="grey"
      disabled={false}
      loading={false}
      name="users" 
    />&nbsp; &nbsp; user02@gmail.com
                    </div>
                       <div className="item" onClick={() => setShowModal(false)}>
                       <Icon
      disabled={true}
      loading={false}
      name="plus"
      size="small"
    />&nbsp; &nbsp;  &nbsp; Add account
                    </div>
                </ModalBody>
              
            </Modal>
        </div>
    );

};

Dialog.defaultProps = {
  open: false,
  fullScreen: false,
  scroll: false,
};




export default Dialog;