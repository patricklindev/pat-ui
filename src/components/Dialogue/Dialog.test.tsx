import React, { useState } from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Dialog, DialogActions, patDialogProps } from './Dialog';

describe('Dialog', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Dialog> Snapshot Dialog </Dialog>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render default dialog', () => {
    const dialogProps: patDialogProps = {
      open: true,
      onClose: jest.fn(),
    };
    const wrapper = render(<Dialog {...dialogProps}></Dialog>);
    const dialogElement = screen.getByTestId('dialog-element');
    expect(dialogElement).toHaveClass('dialog dialog-default');
    const dialogMask = screen.getByTestId('dialog-mask'); // test clicking the mask layer to close
    expect(dialogProps.onClose).toHaveBeenCalledTimes(0);
    fireEvent.click(dialogMask);
    expect(dialogProps.onClose).toHaveBeenCalledTimes(1);
  });

  it('should render dialog with different props', () => {
    const dialogProps: patDialogProps = {
      dialogTitle: 'Title',
      dialogContent: 'I am the content',
      dialogType: 'basic',
      className: 'test',
      open: true,
    };
    const wrapper = render(<Dialog {...dialogProps}></Dialog>);
    const dialogTitleElement = wrapper.queryByText('Title');
    expect(dialogTitleElement).toBeInTheDocument();
    const dialogContentElement = wrapper.queryByText('I am the content');
    expect(dialogContentElement).toBeInTheDocument();
    const Element = screen.getByTestId('dialog-element');
    expect(Element).toHaveClass('dialog dialog-basic test');
  });

  it('should render actions with clickable buttons', () => {
    const dialogProps: patDialogProps = {
      dialogType: 'default',
      open: true,
      onClose: jest.fn(),
    };
    //test button
    const wrapper = render(
      <Dialog {...dialogProps}>
        <DialogActions>
          <button data-testid="btn" onClick={dialogProps.onClose} />
        </DialogActions>
      </Dialog>
    );
    const actionElement = screen.getByTestId('dialog-actions-element'); // test for the action component
    expect(actionElement).toHaveClass('dialog-actions');
    const buttonElement = screen.getByTestId('btn'); // test for the button in the action componenet
    expect(buttonElement.tagName).toBe('BUTTON');
    expect(dialogProps.onClose).toHaveBeenCalledTimes(0); // test the onClose function
    fireEvent.click(buttonElement);
    expect(dialogProps.onClose).toHaveBeenCalledTimes(1);
  });

  it('should render dialog with full-screen type', () => {
    const dialogProps: patDialogProps = {
      dialogType: 'full-screen',
      open: true,
    };
    const wrapper = render(<Dialog {...dialogProps}></Dialog>);
    const dialogElement = screen.getByTestId('dialog-element');
    expect(dialogElement).toHaveClass('dialog dialog-full-screen');
  });
});
