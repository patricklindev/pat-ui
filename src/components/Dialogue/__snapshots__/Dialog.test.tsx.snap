// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component and sub-components should match snapshot 1`] = `<DocumentFragment />`;
